### PR TITLE
Update the package yaml file in kubefed-operator

### PR DESCRIPTION
**Pull request by abhat**
Update package yaml to 4.2.0
Update the image-references file to include ose-kubefed as well. 
Let operator version percolate down through the olm-e2e script.

### CODE REVIEW

**Comment by sohankunkerkar** [comment]
nit: You need to also make changes in this file (https://github.com/openshift/kubefed-operator/blob/master/hack/catalog.sh#L17) to ensure that we're testing our changes against the version listed in `package.yaml` file.

File: scripts/install-kubefed.sh [hunk 1]
```diff
@@ -20,19 +20,21 @@ CSV_PATH="./deploy/olm-catalog/kubefed-operator/${OPERATOR_VERSION}/kubefed-oper
...
 SCOPE="Namespaced"
 
-while getopts “n:d:i:s:” opt; do
+while getopts “n:d:i:s:o:” opt; do
     case $opt in
 	n) NAMESPACE=$OPTARG ;;
 	d) LOCATION=$OPTARG ;;
         i) IMAGE_NAME=$OPTARG ;;
         s) SCOPE=$OPTARG;;
+	o) OPERATOR_VERSION=$OPTARG;;
```
Comment (sohankunkerkar) [inline scripts/install-kubefed.sh:28]: Can you update this information in the README as well?

File: deploy/olm-catalog/kubefed-operator/4.2/image-references [hunk 2]
```diff
@@ -2,6 +2,10 @@ kind: ImageStream
 apiVersion: image.openshift.io/v1
 spec:
   tags:
+  - name: ose-kubefed
```
Comment (sohankunkerkar) [inline deploy/olm-catalog/kubefed-operator/4.2/image-references:5]: So this change is indeed required!?

**Comment by sohankunkerkar** [comment]
Overall, LGTM

File: deploy/olm-catalog/kubefed-operator/4.2/image-references (hunk 2 above)
Comment (abhat) [reply deploy/olm-catalog/kubefed-operator/4.2/image-references:5]: Yes, sorry. I know more now than I did before.

**Pull request merged**